### PR TITLE
convert usleep to nanosleep

### PR DESCRIPTION
**Pull request by neheb**
usleep is removed in POSIX 2008.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: bcmserver.c [hunk 1]
```diff
@@ -179,9 +180,11 @@ int main(void)
 	saddr.sin_addr.s_addr = htonl(INADDR_ANY);
 	saddr.sin_port = htons(PORT);
 
+	struct timespec r;
 	while(bind(sl,(struct sockaddr*)&saddr, sizeof(saddr)) < 0) {
 		printf(".");fflush(NULL);
-		usleep(100000);
+		struct timespec f = {0, 100 * 1000 * 1000};
```
Comment (marckleinebudde) [inline bcmserver.c:186]: Please move the declaration of the struct to the beginning of the scope, please initialize it using C99 initializers.

```
struct timesepc f = {
	.tv_nsec = 100 * 1000 * 1000,
};
```

File: bcmserver.c [hunk 2]
```diff
@@ -179,9 +180,11 @@ int main(void)
...
+		nanosleep(&f, &r);
```
Comment (marckleinebudde) [inline bcmserver.c:187]: You're not using `&r`, so please use `NULL` intead.

**Review by marckleinebudde — CHANGES_REQUESTED** [review]
The comments from the first file applies to the others, too.

File: bcmserver.c (hunk 1 above)
Comment (neheb) [reply bcmserver.c]: Done.

File: bcmserver.c (hunk 2 above)
Comment (neheb) [reply bcmserver.c:190]: I added r as this is in a loop.

Comment (marckleinebudde) [reply bcmserver.c:190]: But `r` is not used, why do you need it?

Comment (neheb) [reply bcmserver.c]: Fixed.

**Review by marckleinebudde — APPROVED** [review]

**Pull request merged**